k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/model/GenericK8sLocatable.kt

```diff
@@ -0,0 +1,22 @@
+package com.amazon.spinnaker.keel.k8s.model
+
+import com.amazon.spinnaker.keel.k8s.*
+import com.netflix.spinnaker.keel.api.Locatable
+import com.netflix.spinnaker.keel.api.SimpleLocations
+
+interface GenericK8sLocatable : Locatable<SimpleLocations> {
+    val metadata: Map<String, String>
+    val template: K8sObjectManifest
+
+    val namespace: String
+        get() = (template.metadata[NAMESPACE] ?: NAMESPACE_DEFAULT) as String
+
+    override val application: String
+        get() = metadata.getValue(APPLICATION).toString()
+
+    override val id: String
+        get() = "$namespace-${template.kind}-${template.metadata[NAME]}".toLowerCase()
+
+    override val displayName: String
+        get() = "$namespace-${template.kind}-${template.metadata[NAME]}".toLowerCase()
```

Comment:
Can we add a Spinnaker account name here? It's used by UI so it would be nice to be able to tell which account this is deployed to.